// coffee-script-brunch/lib/index.js
(function() {
  var CoffeeScriptCompiler, capitalize, coffeescript, formatClassName;

  coffeescript = require('coffee-script');

  capitalize = function(string) {
    return (string[0] || '').toUpperCase() + string.slice(1);
  };

  formatClassName = function(filename) {
    return filename.split('_').map(capitalize).join('');
  };

  module.exports = CoffeeScriptCompiler = (function() {

    CoffeeScriptCompiler.prototype.brunchPlugin = true;

    CoffeeScriptCompiler.prototype.type = 'javascript';

    CoffeeScriptCompiler.prototype.extension = 'coffee';

    CoffeeScriptCompiler.prototype.generators = {
      backbone: (function() {
        var types;
        types = {};
        ['collection', 'model', 'router', 'view'].forEach(function(type) {
          var parent;
          parent = formatClassName(type);
          return types[type] = function(fileName) {
            var className;
            className = formatClassName(fileName);
            return "class exports." + className + " extends Backbone." + parent + "\n";
          };
        });
        return types;
      })()
    };

    function CoffeeScriptCompiler(config) {
      this.config = config;
      null;
    }

    CoffeeScriptCompiler.prototype.compile = function(data, path, callback) {
      var error, result;
      try {
        return result = coffeescript.compile(data);
      } catch (err) {
        return error = err;
      } finally {
        callback(error, result);
      }
    };

    return CoffeeScriptCompiler;

  })();

}).call(this);
